TestClasses/ReadWriteTest.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+import sys
+import yaml
+from BaseTest import BaseTest
+from PerformanceTest import PerformanceTest
+import subprocess
+import os
+import time
+import numpy as np
+from zlib import adler32
+
+class ReadWriteTest(BaseTest, PerformanceTest):
+    def __init__(self, configFile=None, singleEndPoint=None, singleProt:list=None):
+        super().__init__()
+        self.groupedOutput = []
+        self.checksums = []
+
+        if configFile is not None:
+            with open(configFile) as config:
+                try:
+                    sites = yaml.safe_load(config)
+                    self.sitesList = sites['SITES']
+                except yaml.YAMLError as exc:
+                    print(exc)
+        
+        if singleEndPoint is not None:
+            self.sitesList = singleEndPoint
+        if singleProt is not None:
+            self.protocols = singleProt
+        #sys.exit()
+
+    
+    def subprocess(self, action:str, sourcePath=None, destinBaseNm=None, *args, timeout=None):
+        self.setup(action, sourcePath, destinBaseNm, *args)
+        self.groupedOutput = [] # May not need this to be an instance/class variable
+        self.checksums = []
+
+        #Read-write tests will use xrdcp and gfal-copy
+        #If the protocol is root, use xrootd. Otherwise the protocol we use is gfal-copy
+        #create a combination of each protocol, server and root
+
+        if action == 'load': self.sitesList = self.redirect
+        if timeout is not None:
+            timeout = ['timeout', str(timeout)]
+        
+        for site, siteVal in self.sitesList.items(): #Iterate over destination sites
+            for prot in self.protocols: #Iterate over protocols
+                
+                tool = self.cmds[prot] #e.g. tool=xrd
+                endpoint = prot + siteVal[0] + ':' + str(self.port)
+                ext = '_' + prot.strip(':/') + '_' + tool
+                self.destinPath = siteVal[1] + destinBaseNm
+                
+                fullCmd = self.parse(self.basecmd[self.action][tool], self.sourcePath, endpoint, self.destinPath+ext, self.args)
+                if timeout is not None:
+                    fullCmd = timeout + fullCmd
+
+                # print("Full Command:", fullCmd)
+                
+                Testprocess = subprocess.run(fullCmd, capture_output=True, text=True)
+
+                outputs = (Testprocess.returncode, Testprocess.stdout, Testprocess.stderr)
+                self.groupedOutput.append(outputs)
+
+                if action == "checksum":
```

Comment:
same here, if you're only returning checksums when action is set to checksums, you don't need to calculate grouped outputs